client/src/main/java/io/pravega/client/stream/impl/Orderer.java

```diff
@@ -32,13 +33,13 @@ SegmentInputStream nextSegment(List<SegmentInputStream> segments) {
             return null;
         }
         for (int i = 0; i < segments.size(); i++) {
-            SegmentInputStream inputStream = segments.get(counter.incrementAndGet() % segments.size());
+            SegmentInputStream inputStream = segments.get(MathHelpers.abs(counter.incrementAndGet()) % segments.size());
```

Comment:
reaching 2B segments would be huge!
@maddisondavid what is the routing key used while writing events?
Lets say a fixed routing key was used. So all traffic goes to a single segment each time.
Then if a segment is split every 10 minutes, and remainder are merged (as 2 or 3 neighbours) every 20 minutes, to reach 2 billion it will take O(million minutes). 

i dont think overflow would have happened because there are other things that will start breaking at that number, like 
1. the segment table with each record for a segment being 16 bytes. that will make it about 32 gb of a node data in zookeeper. same for history table and index table.
merely reading that from the zk or writing back to zk will take hours.
(the tables are designed for maximum of O(M) segments so that it stays within few mbs of data at worst. 
2. processing through that data, all of it happens in memory --> controller will start getting OOM before we are handing O(B) segments. 

So it would be hard to believe that we had an overflow. things would break much before the overflow would happen.